src/backend/mailbox_manager/models.py

```diff
@@ -25,6 +27,11 @@ class Meta:
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
+        if not self.slug:
+            self.slug = slugify(self.name)
+        return super().save(*args, **kwargs)
```

Comment:
I'm curious about the type of special caracters we would expect in a domain name !